tests/programs/custom/debug_test_trigger/debug_test_trigger.c

```diff
@@ -1416,16 +1471,29 @@ void _debug_mode_register_test(void) {
                        )" ::: "s0", "s1", "s2", "memory");
 
 
-  // TDATA2 (Type==5) - Exception Trigger - Write 1s
-  __asm__ volatile (R"(li   s1,     0xFFFFFFFF
-                       csrw tdata2, s1
-                       csrr s0,     tdata2
-                       li   s1,     0x030009AE
-                       beq  s0,     s1,  1f
-                       li   s1,     0x2   #DEBUG_STATUS_ENTERED_FAIL
-                       sw   s1,     g_debug_entry_status, s2
-                     1:nop
-                       )" ::: "s0", "s1", "s2", "memory");
+  if (has_atomics_configured()) {
+    // TDATA2 (Type==5) - Exception Trigger - Write 1s
+    __asm__ volatile (R"(li   s1,     0xFFFFFFFF
+                         csrw tdata2, s1
+                         csrr s0,     tdata2
+                         li   s1,     0x010008FE
+                         beq  s0,     s1,  1f
+                         li   s1,     0x2   #DEBUG_STATUS_ENTERED_FAIL
+                         sw   s1,     g_debug_entry_status, s2
+                       1:nop
+                         )" ::: "s0", "s1", "s2", "memory");
```

Comment:
I also don't know this feature well enough, so I will simply trust the code.

Reply:
This register has two bits that should only be WARL 0x1, 0x0 if atomics are supported. if A-ext is not supported, these bits should be hardcoded zero (cf. user manual tdata2 for trigger type 5)